tests/lax_numpy_test.py

```diff
@@ -3113,6 +3113,11 @@ def testTraceMethod(self):
     self.assertAllClose(x.trace(), api.jit(lambda y: y.trace())(x),
                         check_dtypes=True)
 
+  def testSquareIsImplementedByMultiply(self):
+    # See https://github.com/google/jax/pull/3036
+    hlo_text = jax.xla_computation(lambda x: jnp.square(x))(42.0).GetHloText()
```

Comment:
Well, I was thinking simpler: just check the numerical property that you cared about (namely that `3. ** 2` is exactly equal to 9.0)

Reply:
It looks better to check the situation with all integers less than [0, 2**12) since float32 can represent exact integers less than 2**24.